velox/exec/RowContainer.h

```diff
@@ -526,6 +611,31 @@ class RowContainer {
     }
   }
 
+  template <typename T>
+  static void extractValuesOffsetsNoNulls(
+      const char* const* rows,
+      const BufferPtr& offsets,
+      int32_t columnOffset,
+      const vector_size_t resultOffset,
+      FlatVector<T>* result) {
+    auto numRows = offsets->size() / sizeof(vector_size_t);
+    auto offsetsVector = offsets->as<vector_size_t>();
+
+    BufferPtr valuesBuffer = result->mutableValues(resultOffset + numRows);
+    auto values = valuesBuffer->asMutableRange<T>();
+    for (int32_t i = 0; i < numRows; ++i) {
+      auto resultIndex = resultOffset + i;
+      auto row = rows[offsetsVector[i]];
+      if (row == nullptr) {
+        result->setNull(resultIndex, true);
+      } else {
+        result->setNull(resultIndex, false);
+        // Here a StringView will reference the hash table, not copy.
```

Comment:
Since there is a specialization for StringViews, this code doesn't run when T is a StringView and therefore the comment doesn't apply, does it?

Reply:
Yeah, this code isn't applicable for StringView. Updated the comment.